feat: test cases for standalone & legacy VPCs (#940)
integr8ly/integreatly-operator: test-cases/tests/installation/a25-verify-standalone-rhmi-vpc-exists-and-is-configured-properly.md

## Prerequisites

- admin access to the AWS account where the OpenShift cluster is provisioned (IAM access key & secret)
- [AWS CLI](https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-install.html) installed locally
- [OCM CLI](https://github.com/openshift-online/ocm-cli/releases) and access to ocm organization where the cluster has been provisioned

## Steps

1. Login via `oc` as a user with **cluster-admin** role (kubeadmin):

```
oc login --token=<TOKEN> --server=https://api.<CLUSTER_NAME>.s1.devshift.org:6443
```

2. Note down CIDR block from `cloud-resources-aws-strategies` config map (search for `_network` field)

```
oc get configmap cloud-resources-aws-strategies -n redhat-rhmi-operator -o yaml
```

3. Get OSD cluster Infra ID and note it down

```
CID=$(ocm get cluster <cluster-id> | jq -r .infra_id)
```

4. List the VPCs in AWS account

```
aws ec2 describe-vpcs --filter "Name=tag-value,Values=$CID"
```

> Verify that the VPC was created and verify that the CIDR block is the same as the one previously noted

5. Following list of verification steps are effectively covered by the functional test

>    > Standalone VPC subnet exists and has the expected CIRD block

Reviewer: These bullet points don't seem to be rendering properly :thinking:

Author: I changed the formatting slightly, hopefully it is more readable now :)

   > Subnet masks for the subnets are one bit greater than the VPC subnet mask

>    > Single VPC security groups with the correct tag exists

Reviewer: Is it worth specifying somewhere what the tag is? These resources are all tagged with `integreatly.org/clusterID=<clusterID>`

- [AWS CLI](https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-install.html) installed locally
- [OCM CLI](https://github.com/openshift-online/ocm-cli/releases) and access to ocm organization where the cluster has been provisioned

## Steps

1. Login via `oc` as a user with **cluster-admin** role (kubeadmin):

```
oc login --token=<TOKEN> --server=https://api.<CLUSTER_NAME>.s1.devshift.org:6443
```

2. Note down CIDR block from `cloud-resources-aws-strategies` config map (search for `_network` field)

```
oc get configmap cloud-resources-aws-strategies -n redhat-rhmi-operator -o yaml
```

3. Get OSD cluster Infra ID and note it down

```
CID=$(ocm get cluster <cluster-id> | jq -r .infra_id)
```

4. List the VPCs in AWS account

```
aws ec2 describe-vpcs --filter "Name=tag-value,Values=$CID"
```

> Verify that the VPC was created and verify that the CIDR block is the same as the one previously noted

5. Following list of verification steps are effectively covered by the functional test
   Note: correct tag for the AWS resources: (Key: `integreatly.org/clusterID`, Value: `<cluster-infra-id>`)

> - Standalone VPC subnet exists and has the expected CIRD block
>

> > - Subnet masks for the subnets are one bit greater than the VPC subnet mask

Reviewer 2: We should ensure there are only 2 subnets, both in a different AZ and both are a private subnet.

>

> > - Single VPC security groups with the correct tag exists

Reviewer 2: Should we call out this security group should have the ingress rules to except all traffic from the `cluster vpc`

>

> > - All RDS subnets exist in a single subnet group

Reviewer 2: I think we should check to ensure there is only 2 subnets in the group, and the subnets match those found earlier.

>

> > - All Elasticache subnets exist in a single subnet group

Reviewer 2: Same as RDS subnet group

>

> > - Single VPC peering connection with the correct tag exists and is active

Reviewer 2: Should we be calling out, to check the peering connection status is active and the requester is the standalone vpc and the accepter is the cluster vpc

>

> > - VPC route table with the correct tag exists and has a route to the peering connection

Reviewer 2: ```suggestion
> - Standalone VPC route table has the correct tag exists and has a route targettiing the peering connection
```
